enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/SpatialFunctionsAcceptanceTest.scala

```diff
@@ -477,6 +477,29 @@ class SpatialFunctionsAcceptanceTest extends ExecutionEngineFunSuite with Cypher
     result.toList should equal(List(Map("point" -> Values.pointValue(CoordinateReferenceSystem.WGS84, 12.78, 56.7))))
   }
 
+  test("Index query with MERGE") {
+    // Given
+    graph.createIndex("Place", "location")
+    graph.execute("CREATE (p:Place) SET p.location = point({latitude: 56.7, longitude: 12.78, crs: 'WGS-84'})")
+    graph.execute("CREATE (p:Place) SET p.location = point({latitude: 56.700001, longitude: 12.7800001, crs: 'WGS-84'})")
+
+    // When matching in merge
+    val result = innerExecuteDeprecated("MERGE (p:Place {location: point({latitude: 56.7, longitude: 12.78, crs: 'WGS-84'}) }) RETURN p.location as point", Map.empty)
```

Comment:
Looks like we are using this deprecated method a lot. Sounds like a bad idea - should we not try to make this new code not build on deprecated stuff?

Reply:
We've been using this method a lot in many places. Clearly a sign that we should rename it and remove the `deprecated` from the name. I think it was named that under the assumption that it should not be necessary to use this method, but in retrospect, it seems very, very useful.